keras/layers/preprocessing/index_lookup.py

```diff
@@ -88,17 +88,17 @@ class IndexLookup(Layer):
 
     def __init__(
         self,
-        max_tokens,
-        num_oov_indices,
-        mask_token,
-        oov_token,
-        vocabulary_dtype,
+        max_tokens=None,
+        num_oov_indices=1,
+        mask_token=None,
+        oov_token="[OOV]",
```

Comment:
I just realized -- the reason there were no default args in this layer is because *there should be no defaults*. This layer does both strings and ints, and it is only used internally. We can't provide defaults if we don't know `vocabulary_dtype`.

Reply:
What about setting max_tokens to None?


I'm seeing this issue now that I've changed it back to 

```
  def __init__(
        self,
        max_tokens,
        num_oov_indices,
        mask_token,
        oov_token,
        vocabulary_dtype,
        vocabulary=None,
        idf_weights=None,
        invert=False,
        output_mode="int",
        sparse=False,
        pad_to_max_tokens=False,
        name=None,
        **kwargs,
    ):
```


```
FAILED keras/layers/preprocessing/index_lookup_test.py::IndexLookupLayerTest::test_ensure_known_vocab_size_without_vocabulary - TypeError: IndexLookup.__init__() missing 1 required positional argument: 'max_tokens'
FAILED keras/layers/preprocessing/index_lookup_test.py::IndexLookupLayerTest::test_idf_weights_set_with_non_tfidf_output_mode - TypeError: IndexLookup.__init__() missing 1 required positional argument: 'max_tokens'
FAILED keras/layers/preprocessing/index_lookup_test.py::IndexLookupLayerTest::test_invert_with_non_int_output_mode - TypeError: IndexLookup.__init__() missing 1 required positional argument: 'max_tokens'
FAILED keras/layers/preprocessing/index_lookup_test.py::IndexLookupLayerTest::test_mask_token_in_wrong_position - TypeError: IndexLookup.__init__() missing 1 required positional argument: 'max_tokens'
FAILED keras/layers/preprocessing/index_lookup_test.py::IndexLookupLayerTest::test_max_tokens_none_with_pad_to_max_tokens - TypeError: IndexLookup.__init__() missing 1 required positional argument: 'max_tokens'
FAILED keras/layers/preprocessing/index_lookup_test.py::IndexLookupLayerTest::test_non_tf_idf_with_idf_weights - TypeError: IndexLookup.__init__() missing 1 required positional argument: 'max_tokens'
FAILED keras/layers/preprocessing/index_lookup_test.py::IndexLookupLayerTest::test_repeated_tokens_in_vocabulary - TypeError: IndexLookup.__init__() missing 1 required positional argument: 'max_tokens'
FAILED keras/layers/preprocessing/index_lookup_test.py::IndexLookupLayerTest::test_sparse_true_with_int_output_mode - TypeError: IndexLookup.__init__() missing 1 required positional argument: 'max_tokens'
FAILED keras/layers/preprocessing/index_lookup_test.py::IndexLookupLayerTest::test_unrecognized_kwargs - TypeError: IndexLookup.__init__() missing 1 required positional argument: 'max_tokens'
FAILED keras/layers/preprocessing/index_lookup_test.py::IndexLookupLayerTest::test_vocabulary_file_does_not_exist - TypeError: IndexLookup.__init__() missing 1 required positional argument: 'max_tokens'
```

Reply:
or shall I fix the tests ? @fchollet

Reply:
OK, I will keep it without changes 
Thank you for your feedback, @fchollet